pkg/types/relayer.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+// PluginArgs are the args required to create any OCR2 plugin components.
+// It's possible that the plugin config might actually be different
+// per relay type, so we pass the config directly through.
+type PluginArgs struct {
+	TransmitterID string
+	PluginConfig  []byte
+}
+
+type RelayArgs struct {
+	ExternalJobID uuid.UUID
+	JobID         int32
+	ContractID    string
+	New           bool // Whether this is a first time job add.
+	RelayConfig   []byte
+}
+
+type PluginRelayer interface {
+	NewRelayer(ctx context.Context, config string, keystore Keystore) (RelayerCtx, error)
+}
+
+// Relayer extends [types.Relayer] and includes [context.Context]s.
+type RelayerCtx interface {
```

Comment:
I dislike the trailing `Ctx` as a long term API signature.

What would it look like to actually remove the depreciated interface `Relayer` and reuse the name here? Any idea how painful that would be for other repos?

how does this type consolidation relate to the other work we need to do -- consolidating the Providers via the proxy and simplifying the ChainStatus, ChainStatuses, NodeStatuses ... and removing ChainStatuses?

that is does the order matter?

Reply:
> What would it look like to actually remove the depreciated interface Relayer and reuse the name here? Any idea how painful that would be for other repos?

I think it would be an easy superficial change, but would have to be coordinated across all of the repos referencing it. 

> that is does the order matter?

I think order is only important for this small `*Ctx` naming issue. For the other stuff, we're already going to be making breaking signature changes in the same module if we don't move them, so I'm not sure it matters :shrug:

Reply:
Updated to drop legacy Relayer to reuse name.

Reply:
thanks. can we hold off on this change until at least tomorrow? i'm trying to organize my thoughts about how to break down the other work. i'd like to minimize the number of cross-repo changes.

Reply:
Yeah no rush on this, and should set up the other related PRs anyways.